pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -361,18 +361,25 @@ func (e *PluginManager) ClearFinalizers(ctx context.Context, o client.Object) er
 	return nil
 }
 
-func (e *PluginManager) Finalize(ctx context.Context, tCtx pluginsCore.TaskExecutionContext) error {
-	// If you change InjectFinalizer on the
-	if config.GetK8sPluginConfig().InjectFinalizer {
-		o, err := e.plugin.BuildIdentityResource(ctx, tCtx.TaskExecutionMetadata())
+func (e *PluginManager) Finalize(ctx context.Context, tCtx pluginsCore.TaskExecutionContext) (err error) {
+	errs := stdErrors.ErrorCollection{}
+	var o client.Object
+	var nsName k8stypes.NamespacedName
+	cfg := config.GetK8sPluginConfig()
+	if cfg.InjectFinalizer || cfg.DeleteResourceOnFinalize {
+		o, err = e.plugin.BuildIdentityResource(ctx, tCtx.TaskExecutionMetadata())
 		if err != nil {
 			// This will recurrent, so we will skip further finalize
 			logger.Errorf(ctx, "Failed to build the Resource with name: %v. Error: %v, when finalizing.", tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName(), err)
 			return nil
 		}
 
 		e.AddObjectMetadata(tCtx.TaskExecutionMetadata(), o, config.GetK8sPluginConfig())
-		nsName := k8stypes.NamespacedName{Namespace: o.GetNamespace(), Name: o.GetName()}
+		nsName = k8stypes.NamespacedName{Namespace: o.GetNamespace(), Name: o.GetName()}
+	}
+
+	// If you change InjectFinalizer on the
```

Comment:
?

Reply:
In my defense, it was like that, I just moved code around :D.... trying to infer intent

Reply:
oh, fair. but probably nice to clean up :)